src/psyclone/domain/common/transformations/invoke_trans.py

```diff
@@ -0,0 +1,238 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2021, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author R. W. Ford STFC Daresbury Lab
+
+'''Specialise generic PSyIR representing an invoke call within the
+algorithm layer to a PSyclone algorithm-layer-specific invoke call
+which uses specialised classes.
+
+'''
+from fparser.two.Fortran2003 import Structure_Constructor
+
+from psyclone.psyir.nodes import Call, ArrayReference, CodeBlock
+from psyclone.psyir.symbols import Symbol, TypeSymbol, StructureType, \
+    RoutineSymbol
+from psyclone.domain.common.algorithm import AlgorithmInvokeCall, \
+    KernelFunctor
+from psyclone.psyGen import Transformation
+from psyclone.psyir.transformations import TransformationError
+from psyclone.psyir.frontend.fparser2 import Fparser2Reader
+
+
+class InvokeTrans(Transformation):
```

Comment:
I believe it is very easy here to get confused with the InvokeSchedule and it would be better to rename this Transformation/filename and associated tests to: InvokeCallTrans.

The same for LFRicInvokeTrans -> LFRicInvokeCallTrans

Reply:
That took a while, but now done.